internal/graph/cursors.go

```diff
@@ -457,7 +457,7 @@ func (ls *parallelLimitedIndexedStream[Q]) forTaskIndex(ctx context.Context, ind
 	// Create a new cursor with cloned limits, because each child task which executes (in parallel) will need its own
 	// limit tracking. The overall limit on the original cursor is managed in completedTaskIndex.
 	childCI := currentCursor.withClonedLimits()
-	childContext, cancelDispatch := branchContext(ctx)
+	childContext, cancelDispatch := context.WithCancelCause(ctx)
```

Comment:
The previous version severed the context completely - did you mean to write WithoutCancel here?

Reply:
I deliberately intended to _not_ sever the context. I couldn't find a reason to justify severed context. This is what prompted this PR: I observed that cancelation wasn't getting propagated to `ReverseQueryRelationship` calls in the data store, which means we could be _not _ freeing_ resources.